epam/async-race.md

```diff
@@ -0,0 +1,206 @@
+- Pagination to display cars (7 per page) and a feature to generate 100 random cars at once.
+
+### Car Animation
+
+- Start/stop engine buttons with corresponding animations and handling of engine states.
+- Adaptive animations that work on screens as small as 500px.
+
+### Race Animation
+
+- A button to start a race for all cars on the current page.
+- A reset button to return all cars to their starting positions.
+- Display the winner's name upon race completion.
+
+### Winners View
+
+- Display winning cars with their image, name, number of wins, and best time.
+- Pagination and sorting capabilities by wins and best times.
+
+## 📜 Non-functional Requirements
+
+- **Framework Usage**:  Development must utilize either React or Angular.
+- **TypeScript Mandatory**: Strong typing throughout, with no use of "any".
+- **Modular Architecture**: Clear separation of concerns between API interaction, UI rendering, and state management.
+- **SPA Implementation**: All content must be generated using TypeScript.
+- **Code Quality**: Adherence to Airbnb's ESLint configuration, with functions limited to 40 lines and avoidance of magic numbers/strings.
+
+- These criteria emphasize the importance of clean code, proper architecture, and adherence to modern development practices. Intreviewers will use these guidelines to assess the technical execution and overall quality of the project.
+
+## 🛠️ Technical Implementation
+
+- Implement CRUD operations for cars using the provided server mock.
+- Design UI elements for car management and race controls.
+- Utilize fetch for server communication, and handle promises for asynchronous tasks.
+- Create animations for car movements using TypeScript and CSS.
+- Ensure responsiveness and compatibility across different devices and browsers.
+
+## Requirements to commits and repo
+### Commit Guidelines
+- Follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) format.
+- Use lowercase for commit types (`init`, `feat`, `fix`, `refactor`, `docs`, etc.).
+- Write in present tense (e.g., "add feature") and imperative mood (e.g., "move cursor to...").
+
+#### Commit Examples
+- `init:` Start a project/task (e.g., `init: start youtube-task`).
+- `feat:` Introduce new features (e.g., `feat: add zoom support`).
+- `fix:` Correct bugs in existing features (e.g., `fix: correct data loading`).
+- `refactor:` Improve the code without altering functionality (e.g., `refactor: rename vars`).
+- `docs:` Update documentation (e.g., `docs: update readme`).
+
+### README.md 
+When your solution is implemented, please include the checklist below in the `README.md`. Check all implemented features, then calculate your score and put it at the top of the `README.md`.
+
+Also, please deploy the UI to gh-pages, AWS Amplify, Netlify, or a similar service, and put the link at the top of the `README.md` (alone with the calculated score).
```

Comment:
let's remove Amplify. It's bad for juniors.

gh-pages, Vercel, Netlify, Cloudflare pages